plugins/uploadimage/plugin.js

```diff
@@ -57,6 +57,10 @@
 				return;
 			}
 
+			if ( !editor.config.imageUploadSupportedTypes ) {
+				editor.config.imageUploadSupportedTypes = /image\/(jpeg|png|gif|bmp|webp)/;
+			}
```

Comment:
Instead of checking the existence of the config variable and setting its value otherwise, you could add the default value to the `CKEDITOR.config` object. You can check how it's done in other plugins, e.g. [`clipboard` one](https://github.com/ckeditor/ckeditor4/blob/ac6a405ce8fe3a43a00ab587bb5d7159aaeac632/plugins/clipboard/plugin.js#L3548-L3555).

Additionally, could you rename the variable to `uploadImage_supportedTypes`? We have a convention that all new config variables are in the `<plugin>_<config variable>` format.